### PR TITLE
Completed express-notes-api challenge - Noah Lett

### PR DESCRIPTION
**Pull request by NoahLett**
image

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const id = Number(req.params.id);
+  if (Number.isInteger(id) === true && id > 0 && Object.hasOwn(data.notes, id)) {
+    delete data.notes[id];
+    fs.writeFile('data.json', JSON.stringify(data, null, 2), 'utf8', err => {
+      if (err) {
+        console.error(err);
+        obj.error = 'An unexpected error occurred.';
+        res.status(500).json(obj);
+      } else {
+        res.status(204).send();
```
Comment (Zhiwutian) [inline express-notes-api/index.js:63]: I would use `sendStatus` here

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
one small updated noted. Also, make sure to update the pull request with a screen capture for all of the functionality. I would recommend creating one capture for each type of end point so they don't get too long. Great work so far!

**Comment by NoahLett** [comment]
[POST a new note.webm](https://user-images.githubusercontent.com/110089434/206259708-8088b912-43ab-431e-8087-8b827d69ec22.webm)

**Comment by NoahLett** [comment]
[POST a new note 500 error.webm](https://user-images.githubusercontent.com/110089434/206259756-1e519c37-482a-4e79-ab51-89bb00d9781d.webm)

**Comment by NoahLett** [comment]
[DELETE note by id 500 error.webm](https://user-images.githubusercontent.com/110089434/206259839-cdb261c0-fd24-4d17-869e-787075c35863.webm)

**Comment by NoahLett** [comment]
[PUT replace a note.webm](https://user-images.githubusercontent.com/110089434/206259896-ba63e41e-59bd-44d0-8d6f-377f31ed7de4.webm)

**Review by Zhiwutian — APPROVED** [review]
Great work!

**Pull request merged**